connectors/kafka/src/main/java/forklift/producers/KafkaForkliftProducer.java

```diff
@@ -0,0 +1,257 @@
+                        "\"doc\":\"Properties added to support forklift interfaces. Format is key,value entries delimited with new lines\"}]}");
+    }
+
+    @Override
+    public String send(String message) throws ProducerException {
+        return sendForkliftWrappedMessage(message, null);
+    }
+
+    @Override
+    public String send(ForkliftMessage message) throws ProducerException {
+        return sendForkliftWrappedMessage(message.getMsg(), message.getProperties());
+    }
+
+    @Override
+    public String send(Object message) throws ProducerException {
+        if (message instanceof SpecificRecord) {
+            return sendAvroMessage((SpecificRecord)message);
+        } else {
+            String json;
+            try {
+                json = mapper.writeValueAsString(message);
+            } catch (JsonProcessingException e) {
+                throw new ProducerException("Error creating Kafka Message", e);
+            }
+            return sendForkliftWrappedMessage(json, null);
+        }
+    }
+
+    @Override
+    public String send(Map<String, String> message) throws ProducerException {
+        return sendForkliftWrappedMessage(this.formatMap(message), null);
+    }
+
+    @Override
+    public String send(Map<Header, Object> headers, Map<String, String> properties, ForkliftMessage message)
+                    throws ProducerException {
+        throw new UnsupportedOperationException("Kafka Producer does not support headers or properties");
+    }
+
+    @Override
+    public String send(Map<String, String> properties, ForkliftMessage message)
+                    throws ProducerException {
+        return this.sendForkliftWrappedMessage(message.getMsg(), properties);
+    }
+
+    @Override
+    public Map<String, String> getProperties() throws ProducerException {
+        return this.properties;
+    }
+
+    @Override
+    public void setProperties(Map<String, String> properties) throws ProducerException {
+        this.properties = properties;
+    }
+
+    @Override
+    public void setHeaders(Map<Header, Object> haeders) {
+        throw new UnsupportedOperationException("Kafka Producer does not support headers or properties");
+    }
+
+    @Override
+    public Map<Header, Object> getHeaders() {
+        throw new UnsupportedOperationException("Kafka Producer does not support headers or properties");
+    }
+
+    @Override
+    public void close() throws IOException {
+        //do nothing, the passed in KafkaProducer may be used elsewhere and should be closed by the KafkaController
+    }
+
+    private String sendForkliftWrappedMessage(String message, Map<String, String> messageProperties) throws ProducerException {
+        GenericRecord avroRecord = new GenericData.Record(forkliftSchema);
+        avroRecord.put(SCHEMA_FIELD_NAME_VALUE, message);
+        messageProperties = messageProperties == null ? new HashMap<>() : new HashMap<>(messageProperties);
```

Comment:
The logic here seems backwards to me. I would think to start with a copy of `properties` and then `putAll` the `messageProperties` on top of it.

Then you can do something like:
```
Map<String, String> appliedProperties = new HashMap<>(properties);

if (messageProperties == null)
    messageProperties = Collections.emptyMap();

appliedProperties.putAll(messageProperties);
```